modApiServer/src/org/aion/api/server/ApiAion.java

```diff
@@ -462,8 +462,10 @@ protected TxRecpt getTransactionReceipt(byte[] txHash) {
     }
 
     protected long estimateNrg(ArgTxCall params) {
-        AionTransaction tx = new AionTransaction(params.getNonce().toByteArray(), params.getFrom(), params.getTo(),
+        Address fromAddr = (params.getFrom().equals(Address.EMPTY_ADDRESS())) ? Address.ZERO_ADDRESS() : params.getFrom();
```

Comment:
Maybe I'm reading this wrong, but isn't this just the same as `Address fromAddr = params.getFrom()` ?

Maybe the condition ternary is supposed to be params.getFrom.isEmpty() ?

Reply:
No, because the EMPTY_ADDRESS and ZERO_ADDRESS are distinct things, so it's not just the same as params.getFrom().

isEmpty() isn't a method in the Address class, as far as I can tell.

Reply:
done